pkg/kubelet/client/kubelet_client.go

```diff
@@ -73,9 +78,20 @@ func MakeTransport(config *KubeletClientConfig) (http.RoundTripper, error) {
 	}
 
 	rt := http.DefaultTransport
-	if config.Dial != nil || tlsConfig != nil {
+	dialer := config.Dial
```

Comment:
I read top to bottom and this construct looks clunky.  this particular instance seems like it may read easier with a wrapper like 

```go
// after the SetOldTransportDefaults
config.Lookup(networkContext).WrapTransport(rt)
```